<a href="https://colab.research.google.com/github/Pradip240/IPL-T20-Cricket-Analysis/blob/main/Ameen_IPL_T20_Cricket_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Indian Premier League(IPL) is a professional Twenty20 cricket league in India contested during March or April and May of every year by eight teams representing eight different cities in India. The league was founded by the Board of Control for Cricket in India(BCCI) in 2008. </b>

## <b> Explore and analyze the data to discover results and statistics for different teams playing in IPL. </b>

##Let's load and explore the dataset

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
#Importing the Numpy & the Pandas libraries.
import numpy as np
import pandas as pd

In [4]:
file_path = '/content/drive/MyDrive/AlmaBetter Data Science/1. PYTHON/CAPSTONE PROJECT/IPL Data/'
ipl_df = pd.read_csv(file_path + 'matches.csv')

In [5]:
#Checking the shape of a dataframe
ipl_df.shape


(756, 18)

In [6]:
#Checking the first five observations
ipl_df.head()

id    Season       city  ...         umpire1        umpire2 umpire3
0   1  IPL-2017  Hyderabad  ...     AY Dandekar       NJ Llong     NaN
1   2  IPL-2017       Pune  ...  A Nand Kishore         S Ravi     NaN
2   3  IPL-2017     Rajkot  ...     Nitin Menon      CK Nandan     NaN
3   4  IPL-2017     Indore  ...    AK Chaudhary  C Shamshuddin     NaN
4   5  IPL-2017  Bangalore  ...             NaN            NaN     NaN

[5 rows x 18 columns]

In [7]:
#Checking the last five observations
ipl_df.tail()

id    Season  ...         umpire2                  umpire3
751  11347  IPL-2019  ...        O Nandan                   S Ravi
752  11412  IPL-2019  ...     Nitin Menon                Ian Gould
753  11413  IPL-2019  ...             NaN                      NaN
754  11414  IPL-2019  ...  Bruce Oxenford  Chettithody Shamshuddin
755  11415  IPL-2019  ...       Ian Gould              Nigel Llong

[5 rows x 18 columns]

In [8]:
#Checking for the null values in the dataset
ipl_df.isnull()

id  Season   city   date  ...  venue  umpire1  umpire2  umpire3
0    False   False  False  False  ...  False    False    False     True
1    False   False  False  False  ...  False    False    False     True
2    False   False  False  False  ...  False    False    False     True
3    False   False  False  False  ...  False    False    False     True
4    False   False  False  False  ...  False     True     True     True
..     ...     ...    ...    ...  ...    ...      ...      ...      ...
751  False   False  False  False  ...  False    False    False    False
752  False   False  False  False  ...  False    False    False    False
753  False   False  False  False  ...  False     True     True     True
754  False   False  False  False  ...  False    False    False    False
755  False   False  False  False  ...  False    False    False    False

[756 rows x 18 columns]

In [9]:
#Checking the info of the datset
ipl_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 756 entries, 0 to 755
Data columns (total 18 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   id               756 non-null    int64 
 1   Season           756 non-null    object
 2   city             749 non-null    object
 3   date             756 non-null    object
 4   team1            756 non-null    object
 5   team2            756 non-null    object
 6   toss_winner      756 non-null    object
 7   toss_decision    756 non-null    object
 8   result           756 non-null    object
 9   dl_applied       756 non-null    int64 
 10  winner           752 non-null    object
 11  win_by_runs      756 non-null    int64 
 12  win_by_wickets   756 non-null    int64 
 13  player_of_match  752 non-null    object
 14  venue            756 non-null    object
 15  umpire1          754 non-null    object
 16  umpire2          754 non-null    object
 17  umpire3          119 non-null    ob

In [16]:
#suming up together the null values for each column
ipl_df.isnull().sum()

id                   0
Season               0
city                 7
date                 0
team1                0
team2                0
toss_winner          0
toss_decision        0
result               0
dl_applied           0
winner               4
win_by_runs          0
win_by_wickets       0
player_of_match      4
venue                0
umpire1              2
umpire2              2
umpire3            637
dtype: int64

##Observations from the dataset:
##1.The dataset has 756 rows and 18 columns.
##2.The dataset hardly has any null values apart from column 'umpire3' which has significant null values. (i.e 637 null values)
##3.So we can drop the 'umpire3' column from the dataframe and proceed further with the EDA. 

In [26]:
#Droping the 'Umpire3' column from the dataset
ipl_df = ipl_df.drop(columns=['umpire3'], axis=1)

In [27]:
ipl_df.head()

id    Season  ...         umpire1        umpire2
0   1  IPL-2017  ...     AY Dandekar       NJ Llong
1   2  IPL-2017  ...  A Nand Kishore         S Ravi
2   3  IPL-2017  ...     Nitin Menon      CK Nandan
3   4  IPL-2017  ...    AK Chaudhary  C Shamshuddin
4   5  IPL-2017  ...             NaN            NaN

[5 rows x 17 columns]

##As our dataset has been cleaned, we can now do the in-depth analysis and visualization.